SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -2307,25 +2293,24 @@ USpatialActorChannel* USpatialNetDriver::CreateSpatialActorChannel(AActor* Actor
 	check(NetConnection != nullptr);
 
 	USpatialActorChannel* Channel = nullptr;
-	if (Actor->GetClass()->HasAnySpatialClassFlags(SPATIALCLASS_Singleton))
+
+	if (Actor->GetClass()->HasAnySpatialClassFlags(SPATIALCLASS_Singleton) && !LoadBalanceStrategy->ShouldHaveAuthority(*Actor))
 	{
-		Channel = GlobalStateManager->AddSingleton(Actor);
+		return Channel;
+	}
+
+	Channel = static_cast<USpatialActorChannel*>(NetConnection->CreateChannelByName(NAME_Actor, EChannelCreateFlags::OpenedLocally));
+	if (Channel == nullptr)
+	{
+		UE_LOG(LogSpatialOSNetDriver, Warning, TEXT("Failed to create a channel for actor %s."), *GetNameSafe(Actor));
 	}
 	else
 	{
-		Channel = static_cast<USpatialActorChannel*>(NetConnection->CreateChannelByName(NAME_Actor, EChannelCreateFlags::OpenedLocally));
-		if (Channel == nullptr)
-		{
-			UE_LOG(LogSpatialOSNetDriver, Warning, TEXT("Failed to create a channel for actor %s."), *GetNameSafe(Actor));
-		}
-		else
-		{
 #if ENGINE_MINOR_VERSION <= 22
-			Channel->SetChannelActor(Actor);
+		Channel->SetChannelActor(Actor);
 #else
-			Channel->SetChannelActor(Actor, ESetChannelActorFlags::None);
+		Channel->SetChannelActor(Actor, ESetChannelActorFlags::None);
 #endif
-		}
 	}
```

Comment:
this looks messy, but it's just a deindentation from being coupled with the if statement that my comment above refers to